src/main/java/seedu/address/logic/commands/slot/EditSlotCommand.java

```diff
@@ -66,6 +66,10 @@ public CommandResult execute(Model model) throws CommandException {
         Slot slotToEdit = lastShownList.get(index.getZeroBased());
         Slot editedSlot = createEditedSlot(slotToEdit, editSlotDescriptor);
 
+        if (!editedSlot.isWithinOneDay()) {
+            throw new CommandException(Messages.MESSAGE_SLOT_NOT_IN_ONE_DAY);
```

Comment:
why is the "not in one day" exception handled in EditSlotCommand for edit slot but in AddSlotParser for add slot?

Reply:
would prefer to perform checks in the parser classes (throw exceptions as early as possible), but since no `Slot` object is created in `EditSlotParser`, checking can only be done in `EditSlotCommand`